goal_src/pc_debug/pc-pad-utils.gc

```diff
@@ -0,0 +1,245 @@
+;;-*-Lisp-*-
+(in-package goal)
+
+
+;; This file is used for debugging and testing the PC port pad (controller/input) implementation.
+;; It contains a function for creating a process for debugging cpad inputs and a function to kill that process.
+;; It also contains a function to start a process to map keys to cpad inputs (X, circle, etc.), and another one to kill it.
+;; THIS FILE IS DEBUG ONLY and will not work if *debug-segment* is not enabled.
+
+;; To run this:
+
+#|
+(make-group "iso")  ;; build the game
+(lt)                ;; connect to the runtime
+(lg)                ;; have the runtime load the game engine
+(test-play)         ;; start the game loop
+(ml "goal_src/pc_debug/pc-pad-utils.gc")  ;; build and load this file.
+|#
+
+(cond (*debug-segment*
+
+;; a structure to hold the handles used in this file
+(deftype pc-pad-proc-list (structure)
+  ((show   handle)
+   (input  handle)
+   )
+  )
+(define *pc-pad-proc-list* (new 'static 'pc-pad-proc-list))
+(set! (-> *pc-pad-proc-list* show) (the handle #f))
+(set! (-> *pc-pad-proc-list* input) (the handle #f))
+
+;; a pc pad process
+(deftype pc-pad-proc (process)
+  ((state-time   seconds)
+   (input-index  uint64)
+   )
+  :heap-base #x10
+  )
+
+(define *pc-pad-button-names*
+    (new 'static 'array string 16
+          "SELECT"
+          "L3"
+          "R3"
+          "START"
+
+          "UP"
+          "RIGHT"
+          "DOWN"
+          "LEFT"
+
+          "L2"
+          "R2"
+          "L1"
+          "R1"
+
+          "TRIANGLE"
+          "CIRCLE"
+          "X"
+          "SQUARE"
+        ))
+
+(defenum pc-pad-input-status
+  (disabled)
+  (enabled)
+  (canceled)
+  )
+
+(defun-debug pc-pad-show-start ()
+  "Start the PC port pad debug display"
+
+  (if (not (handle->process (-> *pc-pad-proc-list* show)))
+    (let ((procp
+      (make-function-process *nk-dead-pool* *active-pool* pc-pad-proc :name 'pc-pad-show
+        (lambda :behavior pc-pad-proc ()
+          (stack-size-set! (-> self main-thread) 512)
```

Comment:
I'm kind of surprised we needed more than the default stack size here.  I'm guessing you hit the limit when this was smaller?